src/Files.App/Helpers/QuickLookHelpers.cs

```diff
@@ -12,9 +12,12 @@ public static class QuickLookHelpers
 
     public static async Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
     {
-        if (!App.AppModel.IsQuickLookSupported || !associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
+        if (!associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
             return;
 
+        if (!App.AppModel.IsQuickLookSupported)
+            await DetectQuickLook();
```

Comment:
Is it better to make `IsQuickLookSupported` nullable so that the `DetectQuickLook` task only runs once?

Reply:
@puppetsw pushed a change for this, can you look it over?

Reply:
We can do it that way. I was thinking if QuickLook was installed while Files was running this would still allow you to use QuickLook without needing to restart Files. - Just means every time we press the spacebar on a folder it does the `DetectQuickLook`

Reply:
The task also runs when files are selected. It's not worth an extra operation to handle an edge case.
For what it's worth, @hez2010 is creating a built-in solution to replace this integration.

Reply:
Pushed some changes to account for that scenario

Reply:
It looks like the event is getting hit three times :thinking:

Reply:
Might have to add the null check back in on the `await DetectQuickLookAvailability();` ?

Reply:
The property is no longer nullable.